### PR TITLE
Scripts that need to be in specific directories are `pushd`ed to the right place.

### PR DESCRIPTION
**Pull request by kendrickt**
### Description
Scripts that parsed PWD or BASH_SOURCE now use `pushd` with `git rev-parse --show-toplevel` and end with a `popd`.

A `set -e` was added to `bin/run-dev` to error early because that one has been prone to errors.

Tested by running the scripts from `universal-applicantion-tool_0.0.1/app`.

### Checklist
- [ ] Created tests which fail without the change (if possible)
- [ ] Extended the README / documentation, if necessary

### Approvals
ALL

### CODE REVIEW

**Comment by kendrickt** [comment]
Is this dependency on git bad?

File: bin/localstack/mk-s3 [hunk 1]
```diff
@@ -1,7 +1,8 @@
 #!/bin/bash
+pushd $(git rev-parse --show-toplevel)
 
 # find repo root name (currently civiform), also support any names for backward and forward compatibility
-CIVIFORM_DIR=$(pwd | awk -F / '{print $(NF-2)}')
+CIVIFORM_DIR=${PWD##*/}
```
Comment (hmhsu) [inline bin/localstack/mk-s3:5]: Suggestion (non-blocking): this is the same as `$(basename $PWD)`

File: bin/run-dev [hunk 2]
```diff
@@ -1,7 +1,10 @@
 #!/bin/bash
+pushd $(git rev-parse --show-toplevel)
+
+set -e
 
 # run bin/pull-image
```
Comment (hmhsu) [inline bin/run-dev:6]: Suggestion (non-blocking): it is clear now the comment is no longer needed.

**Review by hmhsu — APPROVED** [review]
Thanks, definitely an improvement. Only one blocking concern: some of the scripts are run by GitHub in our CI/CD workflow. I don't know if this will break it or not. Please confirm with @ndmckinley that this is okay.

File: bin/build-and-push [hunk 3]
```diff
@@ -1,4 +1,5 @@
 #! /bin/bash
+pushd $(git rev-parse --show-toplevel)
```
Comment (ndmckinley) [inline bin/build-and-push:2]: niiiiice.

**Review by ndmckinley — APPROVED** [review]
Should work!  When you merge, watch the next PR and see if something goes badly wrong, but ... should work!

File: bin/run-dev (hunk 2 above)
Comment (kendrickt) [reply bin/run-dev:6]: duh! i can delete these 😁

File: bin/localstack/mk-s3 (hunk 1 above)
Comment (kendrickt) [reply bin/localstack/mk-s3:5]: done

**Pull request merged**